torch/csrc/jit/api/function_impl.cpp

```diff
@@ -86,6 +92,28 @@ const c10::FunctionSchema& GraphFunction::getSchema() const {
   return *schema_;
 }
 
+ std::shared_ptr<Graph> GraphFunction::optimized_graph() const {
+    std::lock_guard<std::recursive_mutex> lock(compile_mutex);
+    std::shared_ptr<Graph> optimized_graph;
+    std::shared_ptr<Graph>* optimized_graph_ptr = nullptr;
+
+    if (FLAGS_torch_jit_do_not_store_optimized_graph) {
+        optimized_graph  = graph_->copy();
+        optimized_graph_ptr = &optimized_graph;
+    } else {
+      auto& optimized_graph_opt = optimized_graphs_[currentSpecialization()];
+      if (optimized_graph_opt) {
+        return *optimized_graph_opt;
+      }
+      optimized_graph_opt = graph_->copy();
+      optimized_graph_ptr = &optimized_graph_opt.value();
+    }
+    if (getGraphExecutorOptimize()) {
+        preoptimizeGraph(*optimized_graph_ptr, force_no_amp_);
+    }
+    return *optimized_graph_ptr;
+ }
```

Comment:
This makes code a bit shorter (and avoids a rather unusual raw pointer to smart pointer game)
```suggestion
    std::shared_ptr<Graph> optimized_graph;
    std::shared_ptr<Graph>& optimized_graph_ref = &optimized_graph;
    
    if (!FLAGS_torch_jit_do_not_store_optimized_graph) {
      optimized_graph_ref = optimized_graphs_[currentSpecialization()];
       if (optimized_graph_ref) {
         return *optimized_graph_ref;
       }
    } 

    optimized_graph_ref = graph_->copy();

        if (getGraphExecutorOptimize()) {
        preoptimizeGraph(*optimized_graph_ref, force_no_amp_);
    }
    
    return *optimized_graph_ref;
 }
```